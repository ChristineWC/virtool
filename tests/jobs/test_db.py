from virtool.jobs.db import fetch_complete_job


async def test_fetch_complete_job(snapshot, fake2, mongo, static_time, test_job):
    user = await fake2.users.create()
    test_job["user"] = {"id": user.id}

    assert repr(await fetch_complete_job(mongo, test_job)) == snapshot
