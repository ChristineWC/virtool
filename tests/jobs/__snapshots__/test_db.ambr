# name: test_fetch_complete_job
  "Job(id='4c530449', archived=False, created_at=datetime.datetime(2015, 10, 6, 20, 0), progress=1, stage='import_results', state=<JobState.COMPLETE: 'complete'>, user=UserNested(id='bf1b993c', administrator=False, handle='leeashley'), workflow='build_index', acquired=False, args={'name': None, 'username': 'igboyes', 'sample_id': '1e01a382', 'analysis_id': 'e410429b', 'workflow': 'nuvs', 'index_id': '465428b0'}, rights={}, status=[JobStatus(error=None, progress=0, stage=None, state=<JobState.WAITING: 'waiting'>, step_description=None, step_name=None, timestamp=datetime.datetime(2015, 10, 6, 20, 0)), JobStatus(error=None, progress=0, stage=None, state=<JobState.RUNNING: 'running'>, step_description=None, step_name=None, timestamp=datetime.datetime(2015, 10, 6, 20, 0)), JobStatus(error=None, progress=0, stage='mk_analysis_dir', state=<JobState.RUNNING: 'running'>, step_description=None, step_name=None, timestamp=datetime.datetime(2015, 10, 6, 20, 0)), JobStatus(error=None, progress=1, stage='import_results', state=<JobState.COMPLETE: 'complete'>, step_description=None, step_name=None, timestamp=datetime.datetime(2015, 10, 6, 20, 0))], ping=None)"
---
# name: test_processor
  <class 'dict'> {
    'acquired': False,
    'archived': False,
    'args': <class 'dict'> {
      'analysis_id': 'e410429b',
      'index_id': '465428b0',
      'name': None,
      'sample_id': '1e01a382',
      'username': 'igboyes',
      'workflow': 'nuvs',
    },
    'created_at': datetime.datetime(2015, 10, 6, 20, 0),
    'id': '4c530449',
    'key': 'bar',
    'ping': None,
    'progress': 1.0,
    'rights': <class 'dict'> {
    },
    'stage': 'import_results',
    'state': 'complete',
    'status': <class 'list'> [
      <class 'dict'> {
        'error': None,
        'progress': 0,
        'stage': None,
        'state': 'waiting',
        'timestamp': datetime.datetime(2015, 10, 6, 20, 0),
      },
      <class 'dict'> {
        'error': None,
        'progress': 0,
        'stage': None,
        'state': 'running',
        'timestamp': datetime.datetime(2015, 10, 6, 20, 0),
      },
      <class 'dict'> {
        'error': None,
        'progress': 0.091,
        'stage': 'mk_analysis_dir',
        'state': 'running',
        'timestamp': datetime.datetime(2015, 10, 6, 20, 0),
      },
      <class 'dict'> {
        'error': None,
        'progress': 1.0,
        'stage': 'import_results',
        'state': 'complete',
        'timestamp': datetime.datetime(2015, 10, 6, 20, 0),
      },
    ],
    'user': <class 'dict'> {
      'administrator': False,
      'handle': 'leeashley',
      'id': 'bf1b993c',
    },
    'workflow': 'build_index',
  }
---
